src/api/handlers/testimonial.py

```diff
@@ -34,44 +34,40 @@ def registerRoutes(self):
   def info(self, request):
     context: Context = request.context
     args: dict = context.args
-    args = rename_field(args, 'testimonial_id', 'id')
+
+    self.validator.expect("id", int, is_required=True)
+    self.validator.rename("testimonial_id", "id")
+    args, err = self.validator.verify(args)
+
+    if err:
+      return err
+
     testimonial_info, err = self.service.get_testimonial_info(context, args)
     if err:
       return MassenergizeResponse(error=str(err), status=err.status)
     return MassenergizeResponse(data=testimonial_info)
 
+
   @login_required
   def create(self, request):
     context: Context = request.context
     args: dict = context.args
-    args = rename_field(args, 'community_id', 'community')
-    args = rename_field(args, 'action_id', 'action')
-    args = rename_field(args, 'vendor_id', 'vendor')
-    args = rename_field(args, 'preferredName', 'preferred_name')
-    args['tags'] = parse_list(args.get('tags', []))
-
-    # check validity - these should be IDs
-    community = args.get('community', None)
-    if community and not isinstance(community, int):
-        args["community"] = parse_int(community)
-
-    action = args.get('action', None)
-    if action and not isinstance(action, int):
-        args["action"] = parse_int(action)
-
-    vendor = args.get('vendor', None)
-    if vendor and not isinstance(vendor, int):
-        args["vendor"] = parse_int(vendor)
-
-      # To do, if we decide: 
-      # if user specifies other_vendor and passed to API - should record it as an unapproved vendor
-
-    is_approved = args.pop("is_approved", None)
-    if is_approved:
-      args["is_approved"] = parse_bool(is_approved)
-    is_published = args.get("is_published", None)
-    if is_published:
-      args["is_published"] = parse_bool(is_published)
+
+    self.validator.expect("title", str, is_required=True)
```

Comment:
btw you can chain the expects like
```
self.validator
  .expect("title", str, is_required=True)
  .expect('community', int) ......

```

Reply:
@sam - does chaining requests improve speed, or just make it easier coding?